commands/src/main/java/org/jclouds/karaf/commands/compute/ComputeServiceCreateCommand.java

```diff
@@ -145,8 +145,11 @@ public void run() {
                   String providerValue = EnvHelper.getComputeProvider(provider);
                   String apiValue = EnvHelper.getComputeApi(api);
                   String identityValue = EnvHelper.getComputeIdentity(identity);
-                  String credentialValue = EnvHelper.getComputeCredential(credential);
                   String endpointValue = EnvHelper.getComputeEndpoint(endpoint);
+                  String credentialValue = EnvHelper.getComputeCredential(credential);
+                  if (providerValue != null && credentialValue != null && providerValue.equals("google-compute-engine")) {
+                     credentialValue = EnvHelper.getGoogleCredentialFromJsonFile(credentialValue);
```

Comment:
There is no property in the provider. There is a property used in the tests, to denote the json file, but when instantiating the google provider, if you cant to use the json file you need to pass the [GoogleCredentialsFromJson](https://github.com/jclouds/jclouds/blob/master/common/googlecloud/src/main/java/org/jclouds/googlecloud/GoogleCredentialsFromJson.java) credential supplier yourself instead of the credential when creating the context.

>Also, as a minor change, check credentialValue != null first?

Done!